### PR TITLE
Support for settings on Linux and macOS

### LINKED ISSUE #71: YamlCreate Settings for Linux
# Description of the new feature/enhancement

Some users would like to work with Linux OS. Hence, YamlCreate settings should be supported on Linux too.

### PR DESCRIPTION
**Pull request by vedantmgoyal2009**
cc @jedieaston @Trenly
Edit:
- Resolves #71

### CODE REVIEW

**Review by Trenly — COMMENTED** [review]
The path also appears in the documentation

**Comment by jedieaston** [comment]
That's not where that should go on Linux (and probably not on macOS. 

You should be able to use $env:XDG_CONFIG_HOME to get the user's config directory.

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -20,8 +20,15 @@ if ($help) {
     exit
 }
 
+# Set settings directory on basis of Operating System
+# $IsWindows, $IsLinux and $IsMacOS were introduced in PowerShell 6.0+
+# $env:OS for older versions of PowerShell
+if ($IsWindows -or $env:OS) {
+    $script:SettingsPath = Join-Path $env:LOCALAPPDATA -ChildPath 'YamlCreate' # On Windows
+} else {
+    $script:SettingsPath = Join-Path $env:HOME -ChildPath 'YamlCreate' # On Linux and macOS
```
Comment (jedieaston) [inline Tools/YamlCreate.ps1:29]:
```suggestion
    $script:SettingsPath = Join-Path $env:XDG_CONFIG_HOME -ChildPath 'YamlCreate' # On Linux and macOS
```

**Comment by vedantmgoyal2009** [comment]
$env:XDG_CONFIG_HOME doesn't work everywhere.
image

**Review by jedieaston — CHANGES_REQUESTED** [review]
Also, you'll probably want to change some of the prompts to not appear if you aren't running on Windows (i.e. Windows Sandbox isn't a thing on Linux or macOS).

And there's many times where the winget.exe executable is used, which doesn't exist on other OSes either. You'll have to find another way to replace those usages (`winget hash` should be simple, other things might be a little more complicated).

File: Tools/YamlCreate.ps1 (hunk 1 above)
Comment (jedieaston) [reply Tools/YamlCreate.ps1:29]: @vedantmgoyal2009 I see your comment now. 

Maybe if that variable isn't defined, use `$env:HOME/.config/` instead. That's what it's usually set to. XDG variables aren't always set if there isn't a desktop environment installed.

File: Tools/YamlCreate.ps1 [hunk 2]
```diff
@@ -20,8 +20,15 @@ if ($help) {
...
 }
 
+# Set settings directory on basis of Operating System
+# $IsWindows, $IsLinux and $IsMacOS were introduced in PowerShell 6.0+
+# $env:OS for older versions of PowerShell
+if ($IsWindows -or $env:OS) {
+    $script:SettingsPath = Join-Path $env:LOCALAPPDATA -ChildPath 'YamlCreate' # On Windows
+} else {
+    $script:SettingsPath = Join-Path $env:HOME/.config -ChildPath 'YamlCreate' # On Linux and macOS
+}
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:30]:
```suggestion
 $script:SettingsPath = Join-Path $(if ([System.Environment]::OSVersion.Platform -match 'Win') {$env:LOCALAPPDATA} else {$env:HOME+'/.config'} ) -ChildPath 'YamlCreate'
```

**Review by Trenly — COMMENTED** [review]
This can be done on a single line

**Review by Trenly — APPROVED** [review]

**Comment by Trenly** [comment]
I'll wait to see if Easton has anything else to say before merging

**Comment by jedieaston** [comment]
Only thing is that if on line 519, where you get the SignatureSha256 if winget.exe is available, is that value not filled in if you don't have winget? Is there a way to get that value by unzipping the installer ourselves?

**Comment by jedieaston** [comment]
Sounds good to me then.

**Review by jedieaston — APPROVED** [review]
:shipit:

**Pull request merged**